src/mpid/ch4/netmod/ucx/ucx_win.h

```diff
@@ -53,11 +54,21 @@ static inline int MPIDI_UCX_Win_allgather(MPIR_Win * win, size_t length,
      mem_map_params.length = size;
      mem_map_params.flags = 0 ;
 
+    if (*base_ptr == NULL)
```

Comment:
If the user supplies `base_ptr == NULL` do we really need to allocate memory here? Or can we just omit the call to `ucp_mem_map`. Just trying to understand the rationale. I see on line 38, we override the length value to be at least 1K.